lib/index.js

```diff
@@ -12,34 +26,308 @@ class PluginPassportOAuth {
+      .then(userObject => {
+        let req = null;
+
+        if (userObject !== null) {
+          return Promise.resolve(userObject.userId);
+        }
+
+        Object.keys(profile._json).forEach(attr => {
+          if (this.config.strategies[profile.provider].persist.indexOf(attr) > -1) {
+            user[attr] = profile._json[attr];
+          }
+        });
+
+        if (this.config.strategies[profile.provider].persist.length !== 0) {
+          req = {
+            controller: 'security',
+            action: 'createUser',
+            _id: profile._json[this.config.strategies[profile.provider].identifierAttribute],
+            body: {
+              content: {
+                profileIds: this.config.defaultProfiles ? this.config.defaultProfiles : ['default']
+              }, // content will be persisted in Kuzzle
+              credentials: {} // credentials will be persisted in the repository
+            }
+          };
+
+          if (this.config.strategies[profile.provider].kuzzleAttributesMapping) {
+            Object.keys(this.config.strategies[profile.provider].kuzzleAttributesMapping)
+              .forEach(attr => {
+                req.body.content[attr] = profile._json[this.config.strategies[profile.provider].kuzzleAttributesMapping[attr]];
+              });
+          }
+
+          user._id = profile._json[this.config.strategies[profile.provider].identifierAttribute];
+          req.body.credentials[profile.provider] = user;
+
+          return this.context.accessors.execute(new this.context.constructors.Request(request.original, req))
+            .then(() => {
+              return Promise.resolve(profile._json[this.config.strategies[profile.provider].identifierAttribute]);
+            });
+        }
+        return Promise.resolve(profile._json[this.config.strategies[profile.provider].identifierAttribute]);
+      })
+      .catch(err => Promise.reject(err));
+  }
+
+  /**
+   * Check if a given user exists
+   *
+   * @param request
+   * @param userId
+   * @param provider
+   * @returns {Promise|Promise.<boolean>}
+   */
+  exists (request, userId, provider) {
+    return this.getCredentialsFromUserId(userId, provider)
+      .then(credentials => Promise.resolve(credentials !== null));
+  }
+
+  /**
+   * Create a user in the repository of the corresponding strategy
+   *
+   * @param request
+   * @param credentials
+   * @param userId user id in Kuzzle
+   * @param provider
+   * @returns {Promise|Promise.<TResult>}
+   */
+  create (request, credentials, userId, provider) {
+    let user = {};
+
+    return this.exists(request, userId, provider)
+      .then(exists => {
+        if (exists) {
+          return Promise.reject(new this.context.errors.BadRequestError('A strategy already exists for this user.'));
+        }
+        Object.keys(credentials).forEach(attr => {
+          user[attr] = credentials[attr];
+        });
+        user.userId = userId;
```

Comment:
use `user._id = userId` to enable getting user's stored credentials with a simple `Repository.load(userId)` call instead of a `search`